File: app/src/main/java/com/insdn/app/network/NetworkMetrics.java
package com.insdn.app.network;

public class NetworkMetrics {
    private float bandwidth;    // in Mbps
    private float latency;      // in ms
    private float packetLoss;   // in percentage

    public float getBandwidth() { return bandwidth; }
    public void setBandwidth(float bandwidth) { this.bandwidth = bandwidth; }

    public float getLatency() { return latency; }
    public void setLatency(float latency) { this.latency = latency; }

    public float getPacketLoss() { return packetLoss; }
    public void setPacketLoss(float packetLoss) { this.packetLoss = packetLoss; }
} 

File: InSDN-iOS/InSDN/Models/NetworkMetrics.swift
import Foundation

struct NetworkMetrics {
    let bandwidth: Double
    let latency: Double
    let packetLoss: Double
    
    init(bandwidth: Double = 0.0, latency: Double = 0.0, packetLoss: Double = 0.0) {
        self.bandwidth = bandwidth
        self.latency = latency
        self.packetLoss = packetLoss
    }
} 